### PR TITLE
Jenkins configuration

### PR DESCRIPTION
**Pull request by Sybrand**
- Added jenkins templates and documentation

### CODE REVIEW

**Review by Kyubinhan — APPROVED** [review]

File: openshift/jenkins/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- Build the basic Jenkins image. Refer to https://developer.gov.bc.ca, search for: "Migrating-Your-BC-Gov-Jenkins-to-the-Cloud" and see "Building the Jenkins image"
+- Create a primary jenkins server: `oc process -f jenkins.dc.yaml | oc apply -f -`
+- Create a secondary jenkins server: `oc process -f jenkins-secondary.dc.yaml | oc apply -f -`
+
+### Configure jenkins
+
+- Add GitHub credentials to Jenkins. Use type username/password.
+- Add multibranch pipeline.
+- Discover pull requests from origing -> Merging the pull request with the current branch revision.
+- Disover pull requests from forks -> Remove, we don't do forks.
```
Comment (andrea-williams) [inline openshift/jenkins/README.md:24]:
```suggestion
- Discover pull requests from forks -> Remove, we don't do forks.
```

File: openshift/jenkins/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- Build the basic Jenkins image. Refer to https://developer.gov.bc.ca, search for: "Migrating-Your-BC-Gov-Jenkins-to-the-Cloud" and see "Building the Jenkins image"
+- Create a primary jenkins server: `oc process -f jenkins.dc.yaml | oc apply -f -`
+- Create a secondary jenkins server: `oc process -f jenkins-secondary.dc.yaml | oc apply -f -`
+
+### Configure jenkins
+
+- Add GitHub credentials to Jenkins. Use type username/password.
+- Add multibranch pipeline.
+- Discover pull requests from origing -> Merging the pull request with the current branch revision.
```
Comment (andrea-williams) [inline openshift/jenkins/README.md:23]:
```suggestion
- Discover pull requests from origin -> Merging the pull request with the current branch revision.
```

File: openshift/jenkins/jenkins-secondary.dc.yaml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                    path: namespace
+                  - fieldRef:
+                      fieldPath: metadata.labels
+                    name: labels
+                    path: labels
+                  - fieldRef:
+                      fieldPath: metadata.annotations
+                    name: annotations
+                    path: annotations
+            - name: jenkins-slave-user
```
Comment (andrea-williams) [inline openshift/jenkins/jenkins-secondary.dc.yaml:151]: are we back to using the term "slave"?

File: openshift/jenkins/jenkins.dc.yaml [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  - name: MEMORY_LIMIT
+    description: Maximum amount of memory allowed to use
+    displayName: Memory Limit
+    value: 4Gi
+  - description: SLAVE_USER_PASSWORD
+    displayName: SLAVE_USER_PASSWORD
+    from: "[a-zA-Z0-9]{16}"
+    generate: expression
+    name: SLAVE_USER_PASSWORD
+  - description: GitHub Username (Same as github secret)
```
Comment (Kyubinhan) [inline openshift/jenkins/jenkins.dc.yaml:55]: What GitHub credential are we using btw?

**Review by andrea-williams — APPROVED** [review]

File: openshift/jenkins/jenkins-secondary.dc.yaml (hunk 3 above)
Comment (Sybrand) [reply openshift/jenkins/jenkins-secondary.dc.yaml:151]: It looks pretty tightly bound to what Jenkins needs to operate, so I left it alone.

File: openshift/jenkins/jenkins.dc.yaml (hunk 4 above)
Comment (Sybrand) [reply openshift/jenkins/jenkins.dc.yaml:55]: See dev-chatter

File: openshift/jenkins/jenkins-secondary.dc.yaml (hunk 3 above)
Comment (Sybrand) [reply openshift/jenkins/jenkins-secondary.dc.yaml:151]: I'd love to get rid of Jenkins entirely, then the problem goes away...

Comment (andrea-williams) [reply openshift/jenkins/jenkins-secondary.dc.yaml:151]: ah I see

**Pull request merged**